wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/compose/BuildContextImpl.java

```diff
@@ -18,14 +20,18 @@
  * @author Alexander Garagatyi
  */
 public class BuildContextImpl {
-    private String context;
-    private String dockerfile;
+    private String              context;
+    private String              dockerfile;
+    private Map<String, String> args;
 
     public BuildContextImpl() {}
 
-    public BuildContextImpl(String context, String dockerfile) {
+    public BuildContextImpl(String context, String dockerfile, Map<String,String> args) {
         this.context = context;
         this.dockerfile = dockerfile;
+        if (args != null) {
+          this.args = new HashMap<>(args);
+        }
     }
 
     public BuildContextImpl(BuildContextImpl buildContext) {
```

Comment:
I would reuse other existing constructor instead of duplicating constructor content
this(buildContext.getContext(), buildContext.getDockerfile(), buildContext.getArgs())